snapcraft/internal/errors.py

```diff
@@ -454,6 +454,18 @@ def __init__(self, *, elf_file, process_exception, patchelf_version):
                          patchelf_version=patchelf_version)
 
 
+class StagePackageMissingError(SnapcraftError):
+
+    fmt = (
+        '{package!r} is required as a `stage-packages` entry for this part to '
+        'work properly.\n'
+        'Add {package!r} as a stage-packages entry for this part.'
+    )
+
+    def __init__(self, *, package):
```

Comment:
This seems nice and forward-thinking when it comes to removing the automatic crawling (I assume you intend to re-use this exception), however, for that use-case we won't have package names, but library paths. As a result, I only ever see this being used for libc. Perhaps this exception doesn't need to be so generic?

Reply:
the other case that may come in the future is when the new openssl comes in